app/models/vmdb_database_connection.rb

```diff
@@ -25,25 +25,40 @@ class VmdbDatabaseConnection < ApplicationRecord
   virtual_column :blocked_by, :type => :integer
 
   def self.log_statistics(output = $log)
+    log_activity(output)
+    log_table_size(output)
+    log_table_statistics(output)
+  end
+
+  def self.log_csv(keys, stats, label, output)
     require 'csv'
+    csv = CSV.generate do |rows|
+      rows << keys
+      stats.each { |s| rows << s.values_at(*keys) }
+    end
 
-    begin
-      stats = all.map(&:to_csv_hash)
+    output.info("MIQ(#{name}.#{__method__}) <<-#{label}\n#{csv}#{label}")
+  end
 
-      keys = stats.first.keys
+  def self.log_activity(output = $log)
+    stats = all.map(&:to_csv_hash)
+    log_csv(stats.first.keys, stats, "ACTIVITY_STATS_CSV", output)
+  rescue => err
+    output.warn("MIQ(#{name}.#{__method__}) Unable to log activity, '#{err.message}'")
+  end
 
-      csv = CSV.generate do |rows|
-        rows << keys
-        stats.each do |s|
-          vals = s.values_at(*keys)
-          rows << vals
-        end
-      end
+  def self.log_table_size(output = $log)
+    stats = ApplicationRecord.connection.table_size
+    log_csv(stats.first.keys, stats, "TABLE_SIZE_CSV", output)
+  rescue => err
+    output.warn("MIQ(#{name}.#{__method__}) Unable to log activity, '#{err.message}'")
+  end
 
-      output.info("MIQ(#{name}.#{__method__}) <<-ACTIVITY_STATS_CSV\n#{csv}ACTIVITY_STATS_CSV")
-    rescue => err
-      output.warn("MIQ(#{name}.#{__method__}) Unable to log stats, '#{err.message}'")
-    end
+  def self.log_table_statistics(output = $log)
+    stats = ApplicationRecord.connection.table_statistics.sort_by { |h| h['rows_live'] }.reverse!
```

Comment:
Maybe we should just sort by table name and log in slices of 50 or so to avoid the log line limit.

Reply:
So you mean log separate CSVs?

Reply:
Also how often is this logging?  I'm concerned we're going to flood the logs and just cause _other_ problems.

Reply:
yeah, I was considering dropping CSV too

Reply:
This is run every 30 minutes... it's using the existing logging of pg stat activity

`      :db_diagnostics_interval: 30.minutes`

Reply:
Note, we only get approximate row counts by looking at the live/dead rows from autovacuum logging, which we might not capture due to log rotation in the postgresql logs or for tables that haven't vacuumed in the time window of the logging.

Reply:
> This is run every 30 minutes... it's using the existing logging of pg stat activity

I really don't like this going into the main log due to the size and frequency and it's very hard to grep -v it.  Would prefer a separate log, or a dedicated file or something.

Reply:
We only have stdout on podified. Yeah, for grep vs. readability, I don't know.  All I know is we're blind right now.

Reply:
Honestly, we could probably trim out the toasts and indexes but the mechanisms to just exclude them wasn't there so I didn't try that yet.